generator/platformio_generator.py

```diff
@@ -145,4 +145,10 @@
     # Add generated includes and sources to build environment
     #
     env.Append(CPPPATH=[generated_src_dir])
-    env.BuildSources(generated_build_dir, generated_src_dir)
+
+    # Fix for ESP32 ESP-IDF https://github.com/nanopb/nanopb/issues/734#issuecomment-1001544447
+    global_env = DefaultEnvironment()
+    already_called_env_name = "_PROTOBUF_GENERATOR_ALREADY_CALLED_" + env['PIOENV']
```

Comment:
`env['PIOENV']` could contain `-` which are not allowed with scons. So something like `env['PIOENV'].replace("-", "_")` should be there. Otherwise you get a `Illegal construction variable` error. 
See: https://github.com/SCons/scons/blob/master/test/bad-variables.py